builder/virtualbox/common/step_attach_guest_additions.go

```diff
@@ -77,7 +77,7 @@ func (s *StepAttachGuestAdditions) Cleanup(state multistep.StateBag) {
 		"--medium", "none",
 	}
 
-	if err := driver.VBoxManage(command...); err != nil {
-		ui.Error(fmt.Sprintf("Error unregistering guest additions: %s", err))
-	}
+	// Remove the ISO. Note that this will probably fail since
```

Comment:
Quick questions:

```if _, ok := state.GetOk("guest_additions_attached"); ok {```

`ok` would evaluate to true even if we have previously set `state.Put("guest_additions_attached", false)`. Shouldn't the *result* returned by `GetOk` be used here as well/instead?

> I think the complexity introduced by mutating state...

I kind of get where you're coming from, but isn't tracking the changing state through all the steps the point of the state bag? ;)

As always, feedback/explanations helping me to understand all this are much appreciated!